### PR TITLE
Add elasticstack_elasticsearch_security_role_mapping resource

### LINKED ISSUE #47: [Feature] Support for Role Mappings
Having the ability to add and maintain Role Mappings for use with SAML

### PR DESCRIPTION
**Pull request by k-yomo**
Resolves #47

### CODE REVIEW

**Comment by tobio** [comment]
jenkins test this

File: internal/clients/security.go [hunk 1]
```diff
@@ -138,3 +138,60 @@ func (a *ApiClient) DeleteElasticsearchRole(ctx context.Context, rolename string
...
+	defer res.Body.Close()
+	if diags := utils.CheckError(res, "Unable to put role mapping"); diags.HasError() {
+		return diags
+	}
+
+	return nil
+}
+
+func (a *ApiClient) GetElasticsearchRoleMapping(ctx context.Context, roleMappingName string) (*models.RoleMapping, diag.Diagnostics) {
+	req := a.es.Security.GetRoleMapping.WithName(roleMappingName)
```
Comment (tobio) [inline internal/clients/security.go:160]: Can't we chain the WithContext call here?

File: internal/elasticsearch/security/role_mapping.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     true,
+			Description: "Mappings that have `enabled` set to `false` are ignored when role mapping is performed.",
+		},
+		"rules": {
+			Type:             schema.TypeString,
+			Required:         true,
+			DiffSuppressFunc: utils.DiffJsonSuppress,
+			Description:      "A list of mustache templates that will be evaluated to determine the role names that should granted to the users that match the role mapping rules. This matches fields of users, rules can be grouped into `all` and `any` top level keys.",
```
Comment (tobio) [inline internal/elasticsearch/security/role_mapping.go]: This description is for role_templates which is missing from the resource schema.

**Review by tobio — COMMENTED** [review]
We should include role_templates in the resource. Otherwise this looks great, thanks!

**Comment by tobio** [comment]
Also needs a change log entry here :)

File: internal/clients/security.go (hunk 1 above)
Comment (k-yomo) [reply internal/clients/security.go:160]: No, it's not possible...!

**Comment by tobio** [comment]
@k-yomo thanks for the groundwork here. Just wanted to check if you're still able to continue on with this one? If not are you ok with me adding the final changes to support `role_templates` to get this merged?

**Comment by k-yomo** [comment]
@tobio 
I'm not sure when I can get to work on it, so yeah it would be great if you could add. Thank you!!

**Comment by tobio** [comment]
jenkins test this

**Comment by tobio** [comment]
jenkins test this

**Comment by tobio** [comment]
@dimuon are you able to take a look at the changes in this one?

**Comment by dimuon** [comment]
> @dimuon are you able to take a look at the changes in this one?

I'll take a look

**Review by dimuon — APPROVED** [review]
LGTM

File: docs/resources/elasticsearch_security_role_mapping.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "elasticstack_elasticsearch_security_role_mapping Resource - terraform-provider-elasticstack"
+subcategory: ""
```
Comment (calvinbui) [inline docs/resources/elasticsearch_security_role_mapping.md:4]: should be under Security

**Pull request merged**